lib/buffer.js

```diff
@@ -304,6 +306,20 @@ function slowToString(encoding, start, end) {
   if (end > this.length) end = this.length;
   if (end <= start) return '';
 
+  switch (encoding) {
+    case 'ucs2':
+    case 'ucs-2':
+    case 'utf16le':
+    case 'utf-16le':
+      if (((end - start) / 2) > kStringMaxLength)
+        throw new RangeError('invalid String length');
```

Comment:
Shouldn't it be `if (end - start > kStringMaxLength * 2)` or `if (end - start > (kStringMaxLength << 1))`?